### PR TITLE
[REG2.064] Issue 15210 - [ICE] (glue.c at 1489) with tuples and AAs

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=15210

### CODE REVIEW

File: src/func.d [hunk 1]
```diff
@@ -3787,6 +3787,7 @@ extern (C++) Expression addInvariant(Loc loc, Scope* sc, AggregateDeclaration ad
         se = se.semantic(sc);
         se.type = Type.tchar.arrayOf();
         e = new AssertExp(loc, v, se);
+        e.type = Type.tvoid;
```
Comment (ibuclaw) [inline src/func.d]: When is an AssertExp ever not void?

Comment (yebblies) [inline src/func.d]: When it still needs semantic analysis run on it.

Comment (9rnsr) [inline src/func.d]: Ah, I fixed to call `semantic` on the `AssertExp`.

**Comment by 9rnsr** [comment]
Sorry, this is based on `master`, it's incorrect. Replace with #5216.

**Pull request closed without merging**